app/src/main/java/de/tum/in/tumcampusapp/component/other/generic/activity/ProgressActivity.kt

```diff
@@ -94,6 +103,41 @@ abstract class ProgressActivity(
         }
     }
 
+    protected fun showErrorSnackbar(t: Throwable) {
+        val messageResId = when (t) {
+            is UnknownHostException -> R.string.no_internet_connection
+            is InactiveTokenException -> R.string.error_access_token_inactive
+            is InvalidTokenException -> R.string.error_invalid_access_token
+            is MissingPermissionException -> R.string.error_no_rights_to_access_function
+            is TokenLimitReachedException -> R.string.error_access_token_limit_reached
+            is RequestLimitReachedException -> R.string.error_request_limit_reached
+            else -> R.string.error_unknown
+        }
+
+        showErrorSnackbar(messageResId)
+    }
+
+    protected fun showErrorSnackbar(messageResId: Int) {
+        runOnUiThread {
+            Snackbar.make(contentView, messageResId, Snackbar.LENGTH_LONG)
+                    .setAction(R.string.retry) { retryRequest() }
+                    .setActionTextColor(Color.WHITE)
+                    .show()
+        }
+    }
+
+    protected fun showErrorLayout(throwable: Throwable) {
```

Comment:
Snackbars are the superior error display element anyways
Might as well deprecate the error layouts?

Reply:
Nevermind, just read your last commit message :tada: